File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/pat/id/PersonalAccessTokenPermId.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.pat.id;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.id.ObjectPermId;
import ch.systemsx.cisd.base.annotation.JsonObject;

@JsonObject("as.dto.pat.id.PersonalAccessTokenPermId")
public class PersonalAccessTokenPermId extends ObjectPermId implements IPersonalAccessTokenId
{

    private static final long serialVersionUID = 1L;

    public PersonalAccessTokenPermId(String permId)
    {
        super(permId);
    }

    //
    // JSON-RPC
    //

    @SuppressWarnings("unused")
    private PersonalAccessTokenPermId()
    {
        super();
    }

}

File: api-openbis-javascript/src/v3/as/dto/pat/id/PersonalAccessTokenPermId.js
define([ "stjs", "as/dto/common/id/ObjectPermId", "as/dto/pat/id/IPersonalAccessTokenId" ], function(stjs, ObjectPermId, IPersonalAccessTokenId) {
	var PersonalAccessTokenPermId = function(permId) {
		ObjectPermId.call(this, permId);
	};
	stjs.extend(PersonalAccessTokenPermId, ObjectPermId, [ ObjectPermId, IPersonalAccessTokenId ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.pat.id.PersonalAccessTokenPermId';
		constructor.serialVersionUID = 1;
	}, {});
	return PersonalAccessTokenPermId;
})